### PR TITLE
CSV Fix

### PR DESCRIPTION
**Pull request by 00eadeyemi**
### Fixed

- Aggregate Settlement Report in standard CSV format (#159399319)

[Delivers[#159399319](https://www.pivotaltracker.com/story/show/159399319)]

### CODE REVIEW

File: app/Http/Controllers/SettlementsController.php [hunk 1]
```diff
@@ -205,18 +206,28 @@ public function downloadSettlementReportCsv(Merchant $merchant, $id, Request $re
...
+     * @return \Illuminate\Http\Response
+     */
+    private function csvDownload($csv, $filename)
+    {
+        return response()->make(
+            stripcslashes($csv),
+            200,
+            [
+                'Content-Type' => 'text/csv',
+                'Content-Disposition' => 'attachment; filename="'. $filename . '.csv"',
```
Comment (KrekoG) [inline app/Http/Controllers/SettlementsController.php]: cs

File: app/Http/Controllers/SettlementsController.php [hunk 2]
```diff
@@ -105,22 +106,21 @@ public function index(Merchant $merchant, Request $request)
...
-                        $dateRange['date_to']
-                    )
-            );
+            $csvResponse = $this->settlementCsvGateway
+                ->getAggregateSettlementReports(
+                    $merchant->token,
+                    $dateRange['date_from'],
+                    $dateRange['date_to']
+                );
+
```
Comment (EvanBarbour3) [inline app/Http/Controllers/SettlementsController.php]: ...

File: app/Http/Controllers/SettlementsController.php [hunk 3]
```diff
@@ -133,7 +133,8 @@ public function downloadAggregateSettlementReportCsv(Merchant $merchant)
             throw RedirectException::make('/')->setError('Problem fetching Settlements.');
         }
 
-        return $aggregateSettlementReports;
+        return self::csvDownload($csvResponse['csv'], $request->get('filename'));
+
```
Comment (EvanBarbour3) [inline app/Http/Controllers/SettlementsController.php]: ...

Comment (KrekoG) [inline app/Http/Controllers/SettlementsController.php]: cS

File: app/Http/Controllers/SettlementsController.php [hunk 4]
```diff
@@ -105,22 +106,21 @@ public function index(Merchant $merchant, Request $request)
...
-                        $merchant->token,
-                        $dateRange['date_from'],
-                        $dateRange['date_to']
-                    )
-            );
+            $csvResponse = $this->settlementCsvGateway
+                ->getAggregateSettlementReports(
+                    $merchant->token,
+                    $dateRange['date_from'],
+                    $dateRange['date_to']
```
Comment (KrekoG) [inline app/Http/Controllers/SettlementsController.php]: Cs

Comment (KrekoG) [reply app/Http/Controllers/SettlementsController.php]: My bad, it's just a method call.

File: app/Http/Controllers/SettlementsController.php [hunk 5]
```diff
@@ -205,18 +204,28 @@ public function downloadSettlementReportCsv(Merchant $merchant, $id, Request $re
                     throw new \Exception('Settlement report type not found');
             }
 
-            $headers = [
-                'Content-Type' => 'text/csv',
-                'Content-Disposition' => 'attachment; filename="'. $request->get('filename') . '.csv"',
-            ];
+            return self::csvDownload($csvResponse['csv'], $request->get('filename'));
```
Comment (KrekoG) [inline app/Http/Controllers/SettlementsController.php]: this one

**Pull request merged**